app/src/test/java/com/example/parkingsystem/ParkingReservationModelTest.java

```diff
@@ -47,43 +47,68 @@ public void addReservationToParking_reservationAdded_isTrue() {
+        calendar.add(Calendar.DAY_OF_YEAR, 0);
         Date startDate = calendar.getTime();
         calendar.add(Calendar.DAY_OF_YEAR, 1);
-        Date startDate2ndReservation = calendar.getTime();
-        calendar.add(Calendar.DAY_OF_YEAR, 1);
         Date endDate = calendar.getTime();
         calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate2ndReservation = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
         Date endDate2ndReservation = calendar.getTime();
         Assert.assertTrue(model.addReservationToParking(new Reservation("sdj", 2, startDate.getTime(), endDate.getTime())));
-        Assert.assertFalse(model.addReservationToParking(new Reservation("sdj", 2, startDate2ndReservation.getTime(), endDate2ndReservation.getTime())));
+        Assert.assertTrue(model.addReservationToParking(new Reservation("sdj", 2, startDate2ndReservation.getTime(), endDate2ndReservation.getTime())));
     }
 
     @Test
-    public void addReservationToParking_secondReservationTakesPlaceBeforeExistingOneBegins_isFalse() {
+    public void addReservationToParking_secondReservationWithSameArguments_isFalse() {
         Calendar calendar = Calendar.getInstance();
         calendar.add(Calendar.DAY_OF_YEAR, 1);
-        Date startDate2ndReservation = calendar.getTime();
-        calendar.add(Calendar.DAY_OF_YEAR, 1);
-        Date endDate2ndReservation = calendar.getTime();
-        calendar.add(Calendar.DAY_OF_YEAR, 1);
         Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 2);
+        Date endDate = calendar.getTime();
+        Assert.assertTrue(model.addReservationToParking(new Reservation("sdj", 2, startDate.getTime(), endDate.getTime())));
+        Assert.assertFalse(model.addReservationToParking(new Reservation("sdj", 2, startDate.getTime(), endDate.getTime())));
+    }
+
+    @Test
+    public void addReservationToParking_secondReservationBeginsBeforeExistingOneEnds_isFalse() {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, 0);
+        Date startDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date startDate2ndReservation = calendar.getTime();
         calendar.add(Calendar.DAY_OF_YEAR, 1);
         Date endDate = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_YEAR, 1);
+        Date endDate2ndReservation = calendar.getTime();
         Assert.assertTrue(model.addReservationToParking(new Reservation("sdj", 2, startDate.getTime(), endDate.getTime())));
         Assert.assertFalse(model.addReservationToParking(new Reservation("sdj", 2, startDate2ndReservation.getTime(), endDate2ndReservation.getTime())));
     }
+
+    @Test
+    public void addReservationToParking_secondReservationFinishesInTheMiddleOfAnExistingOne_isFalse() {
+
+    }
+
+    @Test
+    public void addReservationToParking_secondReservationTakesPlaceInTheMiddleOfAnExistingOne_isFalse() {
```

Comment:
empty test?

Reply:
Wrote the name but forgot to implement it, it is done now.